docs/HACKING_QUICKSTART.md

```diff
@@ -206,6 +206,7 @@ the amount of disc space used).
 }
 ```
 
+# Todo: A reviewer should have a look on what needs to be done for NixOS with crown now being optional
```

Comment:
I believe this section might be able to just be removed, since `crown` is not generally needed anymore - but I'm not 100% sure, so leaving this section up to reviewers.

Reply:
cc @mukilan

Reply:
I am not sure about this as I don't use VSCode. I guess for developers working on the DOM code, it is useful to have failures from the lint reported in the IDE. In that case I think we should update the documentation in the previous paragraph to pass `--use-crown` to `mach check`.

Reply:
Is `crown` in `PATH` on NixOS? The current instructions tell the user to hardcode the absolute path. (But I'm wondering why this is necessary for vscode, and not in the `.cargo/config.toml`.

Reply:
AFAIK, `crown` will not be in `PATH`. Perhaps @delan will know why VSCode needs this extra configuration.

Reply:
I adjusted the documentation - please review. I mentioned that `cargo build` can be used, but `mach` is still the recommended tool. From the existing documentation I understood that on NixOS `mach` would still be required.

Reply:
Okay, I think I get it now - `vscode` runs outside the nix environment, so `CARGO_BUILD_RUSTC` needs to be set to an absolute path to use `crown`. 
The `mach build` will instead probably use the nix config, so `crown` should then be in path. In that case I guess no documentation updates for the nix part are required.

Reply:
That does make sense to me. But I guess `CARGO_BUILD_RUSTC` will be needed only if `rust-analyzer.check.overrideCommand` is not overridden to use  `mach check` as mach will always enter the nix-shell environment.